src/Front/DesugarStrings.hs

```diff
@@ -17,10 +22,15 @@ desugarStrings (Program decls) = return $ Program $ desugarTop <$> decls
   desugarTop (TopDef d) = TopDef $ desugarDef d
   desugarTop t          = t
 
-  desugarDef (DefFn v bs t e) = DefFn v bs t $ desugarExpr e
-  desugarDef (DefPat b e    ) = DefPat b $ desugarExpr e
+  desugarDef (DefFn v bs t e) = DefFn v bs t $ everywhere (mkT desugarExpr) e
+  desugarDef (DefPat b e    ) = DefPat b $ everywhere (mkT desugarExpr) e
+
 
 -- | Transform a node of type LitString into a node of type ListExpr
 -- For ex, (Lit (LitString "abc")) turns into (ListExpr [97, 98, 99])
 desugarExpr :: Expr -> Expr
-desugarExpr e = e -- TODO: actually implement transformation
+desugarExpr (Lit (LitString s)) = ListExpr (convertList s)
+desugarExpr e                   = e
+
+convertList :: [Char] -> [Expr]
+convertList ls = [ Lit (LitInt (toInteger i)) | i <- map ord ls ]
```

Comment:
Since this func is only called once, could you make it a local func by putting it in a `where` clause inside `desugarExpr`?